core/services/ocr2/plugins/ocr2keeper/evmregistry/v21/logprovider/provider.go

```diff
@@ -289,8 +291,31 @@ func (p *logEventProvider) getLogsFromBuffer(latestBlock int64) []ocr2keepers.Up
 	case BufferVersionV1:
 		// in v1, we use a greedy approach - we keep dequeuing logs until we reach the max results or cover the entire range.
 		blockRate, logLimitLow, maxResults, _ := p.getBufferDequeueArgs()
+
+		windowStart, windowEnd := getBlockWindow(start, blockRate)
+		p.lggr.Debugw("got block window", "windowStart", windowStart, "windowEnd", windowEnd)
+		if p.previousStartWindow != nil {
+			if windowStart != *p.previousStartWindow {
+				p.lggr.Debugw("new block window", "windowStart", windowStart)
+
+				p.currentIteration = 0
+				p.previousStartWindow = &windowStart
+			}
+		}
+
+		if p.currentIteration == 0 {
```

Comment:
I'm just thinking; if a single node in the DON gets restarted, won't the iterations for that node be reset, and cause that node to select a different set of upkeeps compared to other nodes in the DON? Should we somehow incorporate something common across nodes, e.g. the round number, to drive the iteration?